### PR TITLE
PRIME-2572 Add feature to filter HA sites in Admin page

### PR DESCRIPTION
**Pull request by bergomi02**
https://pr-2440.pharmanetenrolment.gov.bc.ca/admin

For testing, some HA Sites are entered under Fraser Health and Interior Health.

### CODE REVIEW

File: prime-angular-frontend/src/app/core/resources/health-authority-resource.service.ts [hunk 1]
```diff
@@ -73,6 +73,25 @@ export class HealthAuthorityResource {
       );
   }
 
+  public getHealthAuthoritySitesByQeury(
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/core/resources/health-authority-resource.service.ts:76]: Typo here

**Comment by neophyte57** [comment]
When I set the State filter, the order of displayed sites changes ... somewhat disorienting:

image

versus 

image

**Comment by neophyte57** [comment]
A lot of Vendors look repeated?  

image

**Comment by neophyte57** [comment]
Care Types also repeated?  Test data issue?  

image

**Comment by bergomi02** [comment]
> When I set the State filter, the order of displayed sites changes ... somewhat disorienting:
> 
> image
> 
> versus
> 
> image

It is updated to sort them by name always.

**Comment by bergomi02** [comment]
> A lot of Vendors look repeated?
> 
> image

It should be fixed.

**Comment by bergomi02** [comment]
> Care Types also repeated? Test data issue?
> 
> image

This is a test data issue. It should not be a problem in Prod.

**Comment by neophyte57** [comment]
I returned to the PR environment and I think a search filter was set from my last visit but the sites weren't filtered appropriately at this point:

image

**Comment by bergomi02** [comment]
> I returned to the PR environment and I think a search filter was set from my last visit but the sites weren't filtered appropriately at this point:
> 
> image

Can't reproduce the issue. The search restored what I searched last time but it works correctly. Maybe some code is still cached in your browser, can you please test again?

**Comment by neophyte57** [comment]
I think for Vendor and Care Type lists, the items should be sorted in case-insensitive manner 

image

File: prime-angular-frontend/src/app/core/resources/health-authority-resource.service.ts [hunk 2]
```diff
@@ -73,6 +73,25 @@ export class HealthAuthorityResource {
...
+    }
+  ): Observable<HealthAuthoritySiteAdminList[]> {
+    const params = this.apiResourceUtilsService.makeHttpParams(queryParam);
+    return this.apiResource.get<HealthAuthoritySiteAdminList[]>(`health-authorities/sites-query`, params)
+      .pipe(
+        map((response: ApiHttpResponse<HealthAuthoritySiteAdminList[]>) => response.result),
+        tap((healthAuthoritySites: HealthAuthoritySiteAdminList[]) => this.logger.info('HEALTH_AUTHORITY_SITES', healthAuthoritySites)),
+        catchError((error: any) => {
+          this.toastService.openErrorToast('Health authority sites could not be retrieved');
+          this.logger.error('[Core] HealthAuthorityResource::getAllHealthAuthoritySites error has occurred: ', error);
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/core/resources/health-authority-resource.service.ts:89]: Copy-and-paste issue in message

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
+import { Component, OnInit, Output, EventEmitter, Input } from '@angular/core';
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts:1]: `Input` not used?

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  public careTypes: Config<number>[];
+
+  private textSearchKey: string = "ha-search-form-textSearch";
+  private siteStatusCodeKey: string = "ha-search-form-siteStatusCode";
+  private vendorCodeKey: string = "ha-search-form-vendorCode";
+  private careTypeCodeKey: string = "ha-search-form-careTypeCode";
+  private assignToMeCodeKey: string = "ha-search-form-assignToMeCode";
+
+  constructor(
+    private route: ActivatedRoute,
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts:41]: `route` not used?

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    return this.form.get('assignToMe') as FormControl;
+  }
+
+  public onRefresh() {
+    this.refresh.emit();
+  }
+
+  public ngOnInit() {
+
+
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts:91]: Extraneous newlines

File: prime-angular-frontend/src/app/core/resources/health-authority-resource.service.ts (hunk 2 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/core/resources/health-authority-resource.service.ts]: fixed

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts (hunk 4 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts]: fixed

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts (hunk 3 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts]: fixed

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts (hunk 5 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts]: fixed

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+      .subscribe((assignToMe: boolean) => {
+        this.localStorage.set(this.assignToMeCodeKey, assignToMe.toString());
+        this.assignToMe.emit(assignToMe);
+      });
+
+    this.form.patchValue({
+      textSearch: this.localStorage.get(this.textSearchKey),
+      siteStatusCode: this.localStorage.getInteger(this.siteStatusCodeKey) || null,
+      vendorCode: this.localStorage.getInteger(this.vendorCodeKey) || null,
+      careTypeCode: this.localStorage.getInteger(this.careTypeCodeKey) || null,
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts:149]: Care Type doesn't seem to be saved/restored properly?  
image
versus
image

**Comment by bergomi02** [comment]
> I think for Vendor and Care Type lists, the items should be sorted in case-insensitive manner
> 
> image

This is fixed.

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/site-registration-tabs/site-registration-tabs.component.ts [hunk 7]
```diff
@@ -47,6 +49,7 @@ export class SiteRegistrationTabsComponent extends AbstractSiteAdminPage impleme
 
   private tabIndexToCareSettingMap: Record<number, CareSettingEnum>;
   private careSettingToTabIndexMap: { [key in CareSettingEnum]?: number };
+  private adminUserId: string;
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/site-registration-tabs/site-registration-tabs.component.ts:52]: Unused now?

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts (hunk 6 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts]: Good catch. It is fixed.

**Review by neophyte57 — APPROVED** [review]
Nicely done!

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/site-registration-tabs/site-registration-tabs.component.ts (hunk 7 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/site-registration-tabs/site-registration-tabs.component.ts]: Removed unused code.

**Pull request merged**